src/Microsoft.AspNet.Razor/Editor/ImplicitExpressionEditHandler.cs

```diff
@@ -18,15 +18,25 @@ namespace Microsoft.AspNet.Razor.Editor
 {
     public class ImplicitExpressionEditHandler : SpanEditHandler
     {
+        private readonly ISet<string> _keywords;
+
         [SuppressMessage("Microsoft.Design", "CA1006:DoNotNestGenericTypesInMemberSignatures", Justification = "Func<T> is the recommended delegate type and requires this level of nesting.")]
         public ImplicitExpressionEditHandler(Func<string, IEnumerable<ISymbol>> tokenizer, ISet<string> keywords, bool acceptTrailingDot)
             : base(tokenizer)
         {
-            Initialize(keywords, acceptTrailingDot);
+            _keywords = keywords ?? new HashSet<string>();
+            AcceptTrailingDot = acceptTrailingDot;
         }
 
-        public bool AcceptTrailingDot { get; private set; }
-        public ISet<string> Keywords { get; private set; }
+        public bool AcceptTrailingDot { get; }
+
+        public IReadOnlyCollection<string> Keywords
```

Comment:
this way we don't have to pay the penalty of querying using only Linq. besides that `IReadOnlyCollection<T>` doesn't support `SetEquals()`.